adagrams/game.py

```diff
@@ -1,11 +1,151 @@
+    "Z":1
+}
+
 def draw_letters():
-    pass
+    pool_list = []
+    chosen_letters = []
+
+    for key in POOL:
+        counter = POOL[key]
+        while counter > 0:
+            pool_list.append(key)
+            counter -= 1
+
+    while len(chosen_letters) < 10:
+        letter = random.choice(pool_list)
+        index = pool_list.index(letter)
+        pool_list.pop(index)
+        chosen_letters.append(letter)
+    
+    return chosen_letters
 
 def uses_available_letters(word, letter_bank):
-    pass
+    letters = []
+    word_cap = word.upper()
+
+    for i in range(0,len(word_cap)):
+        letters.append(word_cap[i])
+
+    count_if_false = 0
+
+    for letter in letters:
+        if letter not in letter_bank:
+            count_if_false += 1
+        elif letters.count(letter) > letter_bank.count(letter):
+            count_if_false += 1
+        else:
+            continue
+
+    if count_if_false == 0:
+        return True
+    else:
+        return False
 
 def score_word(word):
-    pass
+    score = 0
+    word = word.upper()
+    one_point_letters = ["A","E","I","O","U","L","N","R","S","T"]
+    two_point_letters = ["D","G"]
+    three_point_letters = ["B","C","M","P"]
+    four_point_letters = ["F","H","V","W","Y"]
+    five_point_letters = ["K"]
+    eight_point_letters = ["J","X"]
+    ten_point_letters = ["Q","Z"]
+
+    for letter in word:
+        if letter in one_point_letters:
+            score +=1
+        elif letter in two_point_letters:
+            score+=2
+        elif letter in three_point_letters:
+            score+=3
+        elif letter in four_point_letters:
+            score+=4
+        elif letter in five_point_letters:
+            score+=5
+        elif letter in eight_point_letters:
+            score+=8
+        else:
+            score += 10
+    
+    if len(word) >= 7:
+        score += 8
+    
+    return score
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    word_attributes = []
+    word_scores = []
+    winning_word = ""
+    winning_score = 0
+    
+    for word in word_list:
+        word_attributes.append({"word":word,"score":score_word(word),"length":len(word)})
+        word_scores.append(score_word(word))
+
+    
+
+    high_score = max(word_scores)
+
+    words_with_high_score = []
+    
+    
+    #remove any entries that don't have the high score 
+    for i in range(0,len(word_attributes)):
+        if word_attributes[i]["score"] == high_score:
+            words_with_high_score.append(word_attributes[i])
+    
+ 
+    
+    #no tie
+    if len(words_with_high_score) == 1:
+        winning_word = words_with_high_score[0]["word"]
+        winning_score = words_with_high_score[0]["score"]
+    #tie 
+    elif len(words_with_high_score) > 1:
+        word_lengths = []
+        for i in range(0,len(words_with_high_score)):
+            word_lengths.append(words_with_high_score[i]["length"])
+        
+        max_length = max(word_lengths)
```

Comment:
[suggestion] In my experience, matching indices between lists is risky because the lists can get out of sync as code changes. This implementation of `get_highest_word_score` is a little more complex than it could be.

Sometimes when we do these kinds of algorithms that have confusing layers of logical requirements, it's easy to get stuck in a corner by one early-on decision. In this case, I think you got cornered by trying to do this function with the built-in `max` and `min` functions. You'll probably find that if you start rewriting this function from scratch and begin with calculating "max" using a traditional for loop, you'll end up with a simpler, but still correct, result.